### ISSUE TITLE
After Testing model error: name 'samples' is not defined

### ISSUE THREAD

**Issue by TIgorrr** [mozilla/DeepSpeech#2476]
- **Have I written custom code**: No
- **OS Platform and Distribution**: CentOS Linux release 7.7.1908
- **TensorFlow installed from**: installed by pip3, tensorflow-gpu 1.14.0
- **TensorFlow version (use command below)**: 1.14.0-16
- **Python version**: Python 3.6.8
- **Bazel version**:  0.24.1
- **CUDA/cuDNN version**: 10.0 7.5.1
- **GPU model and memory**: 1080 TI
- **DeepSpeech Version**: latest, 0.6.0-alpha.11

My train parameters:
```
python3 -u ~/deepspeech/DeepSpeech/DeepSpeech.py \
  --train_files ~/dataset/train.csv \
  --dev_files ~/dataset/dev.csv \
  --test_files ~/dataset/test.csv \
  --train_batch_size 40 \
  --dev_batch_size 40 \
  --test_batch_size 40 \
  --log_placement true \
  --log_dir ~/deepspeech/logs \
  --test_output_file ~/deepspeech/logs \
  --audio_sample_rate 8000 \
  --n_hidden 500 \
  --epochs 800 \
  --learning_rate 0.0001 \
  --dropout_rate 0.2367 \
  --early_stop true \
  --es_steps 6 \
  --es_mean_th 0.1 \
  --es_std_th 0.1 \
  --use_cudnn_rnn true \
  --export_dir $chkp_dir/model/model_export/ \
  --checkpoint_dir $chkp_dir/model/checkout/ \
  --alphabet_config_path ~/deepspeech/model/alphabet.txt \
  --lm_binary_path ~/deepspeech/model/lm.binary \
  --lm_trie_path ~/deepspeech/model/trie \
  --summary_dir $tfboard_path/tensorboard/ \
  "$@"
```

After testing model step, getting this error:
```
Traceback (most recent call last):
  File "/home/deepuser/deepspeech/DeepSpeech/DeepSpeech.py", line 932, in <module>
    absl.app.run(main)
  File "/home/deepuser/pyenv/deepspeech/lib64/python3.6/site-packages/absl/app.py", line 299, in run
    _run_main(main, args)
  File "/home/deepuser/pyenv/deepspeech/lib64/python3.6/site-packages/absl/app.py", line 250, in _run_main
    sys.exit(main(argv))
  File "/home/deepuser/deepspeech/DeepSpeech/DeepSpeech.py", line 920, in main
    test()
  File "/home/deepuser/deepspeech/DeepSpeech/DeepSpeech.py", line 675, in test
    json.dump(samples, open(FLAGS.test_output_file, 'w'), default=float)
NameError: name 'samples' is not defined

```

**Comment by lissyx** [comment]
It looks like e75d1e4b610670bf40c591915f7488726c8babb8 was broken ? cc @reuben

**Comment by reuben** [comment]
Ah, yes it was. I'll fix it.

**Comment by reuben** [comment]
Fixed in 31ec7a71
